RevenueCatUI/Templates/Components/PackageValidator.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PackageValidator.swift
+//
+//  Created by Josh Holtz on 10/25/24.
+
+import Foundation
+import RevenueCat
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+class PackageValidator {
+
+    private var packageViewModels: [PackageComponentViewModel] = []
+
+    typealias PackageInfo = (package: Package, isDefaultSelected: Bool)
+
+    var packageInfos: [PackageInfo] {
+        self.packageViewModels.compactMap { packageViewModel in
+            guard let package = packageViewModel.package else {
+                return nil
+            }
+            return (package: package, isDefaultSelected: packageViewModel.isDefaultSelected)
+        }
+    }
+
+    func add(_ packageViewModel: PackageComponentViewModel) {
+        self.packageViewModels.append(packageViewModel)
+    }
```

Comment:
Instead of keeping hold of the entire ViewModel, can this function just take a `Package` and an `isSelectedByDefault: Bool`? We don't know how big the ViewModel can grow in the future, and all we need is some data (in the form of `Package` + `Bool`).